src/components/TokenAmount/TokenAmount.js

```diff
@@ -0,0 +1,123 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+import { useToken, UseTokenProvider } from 'use-token'
+import TokenAmount from 'token-amount'
+import { ImageExists } from '../../hooks'
+import { GU, textStyle } from '../../style'
+import { isAddress, warn } from '../../utils'
+import { useTheme } from '../../theme'
+
+const TokenAmountComponent = React.memo(function TokenAmountComponent({
+  address,
+  amount,
+  className,
+  decimals,
+  digits,
+  networkType,
+  size,
+  style,
+  symbol,
+}) {
+  const isValidAddress = isAddress(address)
+  if (!isValidAddress) {
+    warn(`TokenAmount: provided invalid address (${address})`)
+  }
+
+  return (
+    <UseTokenProvider>
+      <div
+        className={className}
+        css={`
+          display: flex;
+          align-items: center;
+        `}
+        style={style}
+      >
+        <Logo address={address} size={size} />
+        {amount && (
+          <span
+            css={`
+              padding-right: ${size === 'large' ? 0.5 * GU : 0.25 * GU}px;
+              ${size === 'large' ? textStyle('title2') : textStyle('body2')};
+              line-height: 1;
+            `}
+          >
+            {new TokenAmount(amount, decimals).format({ digits: digits })}
```

Comment:
We could use the static function since we are only formatting here:

```suggestion
            {TokenAmount.format(amount, decimals, { digits: digits })}
```